File: powerSet.py
import sys

lis=[]
lis=xrange(1,50);
n=len(lis)
max=2**(n)

for i in xrange(max):
	j=i;
	print '{',
	for count in xrange(n):
		if j&1:
			print lis[n-count-1],',',
		j=j>>1
	print '}'

File: powerSet.cpp
#include<iostream>		//Print powerset in fastest way
#include<vector>
#include<cmath>

using namespace std;

int main()
{
 int i,n,j;

 cout<<"\nEnter number of elements: "; cin>>n;
 cout<<"Enter elements of set : ";
 vector<int> set; 
 
 for(i=0; i<n; i++){
   cin>>j;
   set.push_back(j);
 }
 int max=pow(2,n),count;

 cout<<"\nThe power set is :\n"; 

 for(i=0; i<max; i++)
 {
   j=i;
   cout<<"{";
   for(count=1; count<=n; count++)
   {
     if(j&1)
	cout<<set[n-count]<<", ";
     j=j>>1;
   }
  cout<<"}"<<endl;
 }
}
